### DISCUSSION TITLE
Exit this page — we want to know your use cases

### DISCUSSION THREAD

**Discussion by calvin-lau-sig7** [alphagov/govuk-design-system#2905] [Exit this page]
We're published a component to help users [quickly exit a service](https://design-system.service.gov.uk/components/exit-this-page/) and an [accompanying pattern](https://design-system.service.gov.uk/patterns/exit-a-page-quickly/). We’d welcome your feedback on how it’s being used in services.

## What we’re looking for
Use cases (the current component is based in research with people with a lived experience of domestic abuse)

**Comment by Lisa-thompson3** [comment]
**Feedback on the 'Exit this page' Component**
**Context**
Department: Ministry of Justice (MoJ)

User Groups: Domestic abuse survivors and accessibility users (specifically screen reader and keyboard users).

**Identified Issues**
Our research identified two critical friction points that compromise the safety and usability of the component:

1. Discovery and DOM Position
The button is currently announced as the final element on the page. For screen reader users, this means the safety feature is only discovered at the end of the page content. This feels counter-intuitive and potentially dangerous in high-stress contexts.

User Insight:  "[Exit button being read after the continue button] Why would you go past the continue button?" > — IDI 7 (Screen reader user)

2. Shortcut Conflict (Triple-Shift)
The triple-press Shift key activation frequently conflicts with standard screen reader navigation. Because Shift is used in common commands (e.g., Shift + Tab), users triggered the "Exit" function accidentally, causing them to lose their place in the service.

**Tested Improvements & Recommendations**
We modified the component to improve discovery and prevent accidental triggers. These changes tested successfully with our user groups:

Initial Announcement: We added visually hidden (accessible) text at the very beginning of each page: "To exit this page, press escape." This ensures users are aware of the safety exit immediately upon landing, rather than finding it at the footer.

Trigger Key Change: We remapped the activation key to Escape.

**The Case for Standardization**
We recommend GDS considers updating the component to use the Escape key as a standard for the following reasons:

Mental Models: The "Escape" key is globally recognized as the "get me out of here" or "cancel" command. Mapping a safety exit to this key aligns with existing user behavior.

Reduced Interference: Unlike the Shift key, the Escape key is rarely used as a modifier for complex screen reader shortcuts, significantly reducing the risk of accidental activation.

Proactive Accessibility: Moving the instruction to the start of the page (via hidden text) transforms the component from a reactive element to a proactive safety feature for those using assistive technology.
